client/src/components/bonuses.jsx

```diff
@@ -1,12 +1,51 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import $ from 'jquery';
+import BonusItem from './BonusItem.jsx';
 
-const Bonuses = (props) => {
-  return (
-    <div>
-      bonuses
-    </div>
-  );
+class Bonuses extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      bonusTitle: 'Bonus',
+      bundleNumber: null, //number
+      bonus: { bonus_info: [{ cover: '' }] }, //object
+      selectedSong: '', //string, representing url
+    };
+
+    this.selectSong = this.selectSong.bind(this);
+  }
+
+  componentDidMount() {
+    let bundleId = parseInt(window.location.pathname.slice(6));
+    console.log(bundleId)
+
+    $.get(`/bonus/${bundleId}`, (data) => {
+      console.log(data);
+      this.setState({ bonus: data[0] });
+      if (data[0].bonus_info.length > 1) {
+        this.setState({ bonusTitle: 'Bonuses' })
+      }
+    });
```

Comment:
Don't forget to do some error handling here. What happens if `data` doesn't have a property `[0]`? Or if `data[0]` doesn't have a property `.bonus_info`?

Reply:
0 index issue noted for future error handling in PR; `bonus_info` will be caught early by error handling coming out of my DB.